pilot/pkg/kube/inject/inject.go

```diff
@@ -555,21 +561,28 @@ func InjectionData(sidecarTemplate, valuesConfig, version string, deploymentMeta
 		"contains":            flippedContains,
 	}
 
-	var tmpl bytes.Buffer
-	temp := template.New("inject")
-	t, err := temp.Funcs(funcMap).Parse(sidecarTemplate)
+	bbuf, err := parseTemplate(valuesConfig, funcMap, data)
```

Comment:
I am not sure its a good idea to template the full values config, it may be too complex. What we could do is just add a `template` function to https://github.com/istio/istio/blob/a9ca5b84a83129019a4f01f7770068543c20d32b/pilot/pkg/kube/inject/inject.go#L540 and then where we use the PodDNS stuff just use `{{ template .PodDns }}` or something similar

Reply:
Rendering the whole value template could be expensive. I will add a separate function to render part of the value config.